File: SwiftGameOfLife/GameOfLife/BoardSIze.swift
//
//  BoardSIze.swift
//  GameOfLife
//
//  Created by Eirik Vale Aase on 02/04/2022.
//

struct BoardSize {
    let width: Int
    let height: Int

    init(width: Int, height: Int) {
        self.width = width
        self.height = height
    }

    init(size: Int) {
        self.width = size
        self.height = size
    }
}

File: PythonGameOfLife/BoardSize.py
class BoardSize:
    def __init__(self, width: int, height: int):
        self.width = width
        self.height = height

    @classmethod
    def with_width_and_height(cls, width: int, height: int):
        cls(width=width, height=height)

    @classmethod
    def width_size(cls, size: int):
        cls(width=size, height=size)
